core/src/main/java/com/graphhopper/search/EdgeKVStorage.java

```diff
@@ -197,127 +233,198 @@ public Map<String, String> getAll(final long entryPointer) {
+            map.put(key, obj);
         }
 
-        // value for specified key does not existing for the specified pointer
         return map;
     }
 
-    private int putIntoMap(Map<String, String> map, long tmpPointer, int currentKeyIndex) {
-        int valueLength = vals.getByte(tmpPointer) & 0xFF;
-        tmpPointer++;
-        String valueStr;
-        if (valueLength == 0) {
-            valueStr = "";
-        } else {
+    private boolean hasDynLength(Class<?> clazz) {
+        return clazz.equals(String.class) || clazz.equals(byte[].class);
+    }
+
+    private int getFixLength(Class<?> clazz) {
+        if (clazz.equals(Integer.class) || clazz.equals(Float.class)) return 4;
+        else if (clazz.equals(Long.class) || clazz.equals(Double.class)) return 8;
+        else throw new IllegalArgumentException("unknown class " + clazz);
+    }
+
+    private byte[] getBytesForValue(Class<?> clazz, Object value) {
+        byte[] bytes;
+        if (clazz.equals(String.class)) {
+            bytes = ((String) value).getBytes(Helper.UTF_CS);
+        } else if (clazz.equals(Integer.class)) {
+            return bitUtil.fromInt((int) value);
+        } else if (clazz.equals(Long.class)) {
+            return bitUtil.fromLong((long) value);
+        } else if (clazz.equals(Float.class)) {
+            return bitUtil.fromFloat((float) value);
+        } else if (clazz.equals(Double.class)) {
+            return bitUtil.fromDouble((double) value);
+        } else if (clazz.equals(byte[].class)) {
+            bytes = (byte[]) value;
+        } else
+            throw new IllegalArgumentException("value class not supported " + clazz.getSimpleName());
+        if (bytes.length > MAX_LENGTH)
+            throw new IllegalArgumentException("bytes.length cannot be > " + MAX_LENGTH + " but was " + bytes.length);
+        return bytes;
+    }
+
+    private String classToShortName(Class<?> clazz) {
+        if (clazz.equals(String.class)) return "S";
+        else if (clazz.equals(Integer.class)) return "i";
+        else if (clazz.equals(Long.class)) return "l";
+        else if (clazz.equals(Float.class)) return "f";
+        else if (clazz.equals(Double.class)) return "d";
+        else if (clazz.equals(byte[].class)) return "[";
+        else throw new IllegalArgumentException("Cannot find short name. Unknown class " + clazz);
+    }
+
+    private Class<?> shortNameToClass(String name) {
+        if (name.equals("S")) return String.class;
+        else if (name.equals("i")) return Integer.class;
+        else if (name.equals("l")) return Long.class;
+        else if (name.equals("f")) return Float.class;
+        else if (name.equals("d")) return Double.class;
+        else if (name.equals("[")) return byte[].class;
+        else throw new IllegalArgumentException("Cannot find class. Unknown short name " + name);
+    }
+
+    private static class LengthResult {
+        int len;
+    }
```

Comment:
Why is this needed? Why no primitive int or Integer?

Reply:
Ah now I get it. You need a mutable object that you can use as result as method argument. But if you need to return multiple values why do you not create a result class that contains all the result values? Especially considering that you created a result class already :) Anyway, not so important.